pinot-common/src/main/java/org/apache/pinot/common/config/TableConfig.java

```diff
@@ -118,14 +118,24 @@ public static TableConfig fromJsonString(String jsonString)
   @Nonnull
   public static TableConfig fromJSONConfig(@Nonnull JsonNode jsonConfig)
       throws IOException {
+    // Mandatory fields
+    Preconditions.checkState(jsonConfig.has(TABLE_TYPE_KEY), "Table type is missing");
     TableType tableType = TableType.valueOf(jsonConfig.get(TABLE_TYPE_KEY).asText().toUpperCase());
+    Preconditions.checkState(jsonConfig.has(TABLE_NAME_KEY), "Table name is missing");
     String tableName = TableNameBuilder.forType(tableType).tableNameWithType(jsonConfig.get(TABLE_NAME_KEY).asText());
-
+    Preconditions
+        .checkState(jsonConfig.has(VALIDATION_CONFIG_KEY), "Mandatory config '%s' is missing", VALIDATION_CONFIG_KEY);
```

Comment:
Since `"Mandatory config '%s' is missing"` is commonly used here, can we put it as a final static variable in the same class?

Reply:
Done